### PR TITLE
fix: hybrid neural network deprecated package fix #2062

### LINKED ISSUE #2062: Bug: Cannot run the quiskit aer simulator
### Where does the issue happen?

https://learn.qiskit.org/course/ch-applications/hybrid-quantum-classical-neural-networks-with-pytorch-and-qiskit

### What is the content issue?

I was running the code locally and came across an error saying that the example uses a deprecated package in the defined quantum circuit. The circuit is using a qobj for run() is deprecated as of qiskit-aer 0.9.0.

### Screenshots

image

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by nuvinga**
## Changes

Fixes the usage of deprecated q_obj for run() in `machine-learning-qiskit-pytorch.ipynb`.
Closes #2062 

## Implementation details

Introduces changes to the `machine-learning-qiskit-pytorch.ipynb` notebook by removing the `qobj` statement and replaced with `t_qc`. Using a `qobj` for `run()` is deprecated as of qiskit-aer 0.9.0 and transpiled circuits should now be passed directly using `backend.run(circuits, **run_options)`.

The change has been made in both the `v1` and `v2` notebooks.

## Screenshots

FIle paths: `notebooks/v1/ch-machine-learning/machine-learning-qiskit-pytorch.ipynb` and `notebooks/v2/ch-machine-learning/machine-learning-qiskit-pytorch.ipynb`

image

### CODE REVIEW

**Comment by vabarbosa** [comment]
FYI @frankharkins

File: notebooks/v1/ch-machine-learning/machine-learning-qiskit-pytorch.ipynb [hunk 1]
```diff
@@ -160,10 +160,12 @@
...
-    "        qobj = assemble(t_qc,\n",
-    "                        shots=self.shots,\n",
-    "                        parameter_binds = [{self.theta: theta} for theta in thetas])\n",
-    "        job = self.backend.run(qobj)\n",
+    "        # Using a qobj for run() is deprecated as of qiskit-aer 0.9.0.\n",
+    "        # Therefore the transpiled qc (t_qc) is directly passed to the run method.\n",
+    "        # qobj = assemble(t_qc,\n",
+    "        #                 shots=self.shots,\n",
+    "        #                 parameter_binds = [{self.theta: theta} for theta in thetas])\n",
+    "        job = self.backend.run(t_qc)\n",
```
Comment (frankharkins) [inline notebooks/v1/ch-machine-learning/machine-learning-qiskit-pytorch.ipynb:168]: Does this work correctly without the `shots` and `parameter_binds` arguments?

File: notebooks/v1/ch-machine-learning/machine-learning-qiskit-pytorch.ipynb [hunk 2]
```diff
@@ -160,10 +160,12 @@
...
     "                         self.backend)\n",
-    "        qobj = assemble(t_qc,\n",
-    "                        shots=self.shots,\n",
-    "                        parameter_binds = [{self.theta: theta} for theta in thetas])\n",
-    "        job = self.backend.run(qobj)\n",
+    "        # Using a qobj for run() is deprecated as of qiskit-aer 0.9.0.\n",
+    "        # Therefore the transpiled qc (t_qc) is directly passed to the run method.\n",
+    "        # qobj = assemble(t_qc,\n",
+    "        #                 shots=self.shots,\n",
+    "        #                 parameter_binds = [{self.theta: theta} for theta in thetas])\n",
```
Comment (frankharkins) [inline notebooks/v1/ch-machine-learning/machine-learning-qiskit-pytorch.ipynb:167]:
```suggestion
```

You can remove the comment as the user doesn't need to see the old code.

**Review by frankharkins — CHANGES_REQUESTED** [review]
@nuvinga thanks! Can you make this PR to [Qiskit/textbook](https://github.com/Qiskit/textbook) instead please? Just copy this notebook across with the changes.

File: notebooks/v1/ch-machine-learning/machine-learning-qiskit-pytorch.ipynb (hunk 1 above)
Comment (nuvinga) [reply notebooks/v1/ch-machine-learning/machine-learning-qiskit-pytorch.ipynb:168]: its does, the expected output was derived

File: notebooks/v1/ch-machine-learning/machine-learning-qiskit-pytorch.ipynb (hunk 2 above)
Comment (nuvinga) [reply notebooks/v1/ch-machine-learning/machine-learning-qiskit-pytorch.ipynb:167]: will do

**Comment by nuvinga** [comment]
Hi @frankharkins, I've made the changes to the textbook and the following is the PR:

[#52](https://github.com/Qiskit/textbook/pull/52)

**Comment by nuvinga** [comment]
Hi @frankharkins & @vabarbosa. just checking up on this.

**Pull request closed without merging**